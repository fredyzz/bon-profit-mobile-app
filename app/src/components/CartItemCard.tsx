import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {DishGroup} from '../store/context/RestaurantContext/interfaces';
import Icon from 'react-native-vector-icons/Ionicons';
import {globalStyles} from '../theme/appTheme';
import {cardComponent} from '../theme/cardComponent';
import {Colors} from '../theme/colors';

interface Props {
  dishGroup: DishGroup;
  add: () => void;
  remove: () => void;
}
export const CartItemCard = ({dishGroup, add, remove}: Props) => {
  return (
    <View style={styles.container}>
      <View style={styles.dishData}>
        <Text
          style={
            styles.title
          }>{`${dishGroup.quantity} x ${dishGroup.dish.title}`}</Text>
        <Text style={styles.category}>{dishGroup.dish.category}</Text>
        <Text style={styles.price}>â‚¬{dishGroup.dish.price}</Text>
      </View>
      <View style={styles.actionsContainer}>
        <TouchableOpacity
          style={[cardComponent.actionButton, cardComponent.actionButtonLeft]}
          onPress={remove}>
          <Icon
            name="remove-outline"
            style={[globalStyles.icon, cardComponent.actionButtonText]}
          />
        </TouchableOpacity>
        <TouchableOpacity
          style={[cardComponent.actionButton, cardComponent.actionButtonRight]}
          onPress={add}>
          <Icon
            name="add-outline"
            style={[globalStyles.icon, cardComponent.actionButtonText]}
          />
        </TouchableOpacity>
      </View>
      <View />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 14,
    marginLeft: 5,
    backgroundColor: Colors.white,
    height: 100,
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 20,
    borderBottomWidth: 1,
    borderBottomColor: Colors.primary,
  },
  actionsContainer: {
    flexDirection: 'row',
    paddingRight: 20,
  },
  dishData: {
    marginLeft: 20,
    height: '100%',
    width: 160,
    justifyContent: 'center',
  },
  title: {
    color: Colors.dark,
    fontSize: 18,
  },
  category: {
    color: Colors.resalt,
    fontSize: 14,
    marginTop: 2,
  },
  price: {
    color: Colors.dark,
    fontSize: 14,
    marginTop: 2,
  },
});
