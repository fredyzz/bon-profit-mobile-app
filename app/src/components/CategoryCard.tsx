import * as React from 'react';
import {Text, View, StyleSheet, TouchableOpacity} from 'react-native';
import {Colors} from '../theme/colors';

interface CardProps {
  category: string;
  callBack: (category: string) => void;
  isSelected: boolean | undefined;
}

export const CategoryCard = ({category, callBack, isSelected}: CardProps) => (
  <View style={[styles.categoryCard, isSelected && styles.selected]}>
    <TouchableOpacity
      onPress={() => callBack(category)}
      testID={`categoryCard-${category}`}>
      <Text style={isSelected && styles.selected}>{category}</Text>
    </TouchableOpacity>
  </View>
);

const styles = StyleSheet.create({
  categoryCard: {
    width: 120,
    height: 120,
    backgroundColor: Colors.white,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.2,
    shadowRadius: 3.84,

    elevation: 4,
    margin: 10,
    borderRadius: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  selected: {
    backgroundColor: Colors.primary,
    color: Colors.white,
  },
});
