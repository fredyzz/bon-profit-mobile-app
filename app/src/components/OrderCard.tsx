import React from 'react';
import {StyleSheet, Text, View, Image} from 'react-native';
import {Colors} from '../theme/colors';
import {Order} from '../store/context/OrdersContext/interfaces';
import {Dish} from '../store/context/RestaurantContext/interfaces';
import {getLongestTime} from '../helpers/order.helper';
import Icon from 'react-native-vector-icons/Ionicons';

interface Props {
  order: Order;
}
export const OrderCard = ({order}: Props) => {
  const orderDate = new Date(order.date);
  const formatedDate = `${orderDate.getDay()}/${orderDate.getMonth()}/${orderDate.getFullYear()}`;
  const formatedTime = `${orderDate.getHours()}:${orderDate.getMinutes()}`;
  return (
    <View style={styles.container}>
      <View style={styles.dishesImagesContainer}>
        {order.dishes.slice(0, 4).map((dish: Dish, index: number) => (
          <Image
            source={{uri: dish.imagesHref[0]}}
            style={styles.dishImage}
            key={index}
          />
        ))}
        {order.dishes.length > 5 && (
          <View style={[styles.dishImage, styles.dishPlaceholder]}>
            <Icon
              name="ellipsis-horizontal"
              style={styles.dishPlaceholderIcon}
            />
          </View>
        )}
      </View>
      <View style={styles.dishData}>
        <Text style={styles.title}>
          Ordered: {formatedDate} - {formatedTime}
        </Text>
        {!order.isDelivered && (
          <Text style={styles.title}>
            Waiting estimated time: {getLongestTime(order.dishes)}
          </Text>
        )}

        <Text style={styles.stateText}>
          {order.isDelivered ? 'Delivered' : 'Waiting for delivery'}
        </Text>
      </View>
      <View />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: 300,
    minHeight: 200,
    margin: 10,
    alignItems: 'center',
    justifyContent: 'space-around',
    backgroundColor: Colors.white,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.2,
    shadowRadius: 3.84,

    elevation: 4,
    borderRadius: 20,
  },
  actionsContainer: {
    flexDirection: 'row',
    paddingRight: 20,
  },
  dishData: {
    marginLeft: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: Colors.primary,
    fontSize: 18,
    letterSpacing: 1,
  },
  stateText: {
    marginTop: 14,
    color: Colors.dark,
    fontSize: 18,
    letterSpacing: 1,
  },
  dishesImagesContainer: {
    flexDirection: 'row',
    height: 50,
  },
  dishImage: {
    width: 60,
    height: 60,
    borderRadius: 100,
    borderWidth: 2,
    borderColor: Colors.primary,
    backgroundColor: Colors.white,
    marginRight: -10,
  },
  dishPlaceholder: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  dishPlaceholderIcon: {
    fontSize: 26,
  },
});
