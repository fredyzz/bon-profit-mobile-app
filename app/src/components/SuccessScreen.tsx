import {DrawerScreenProps} from '@react-navigation/drawer';
import React, {useEffect, useContext} from 'react';
import {AuthContext} from '../store/context/AuthContext';
import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableOpacity,
  SafeAreaView,
} from 'react-native';
import {globalStyles} from '../theme/appTheme';
import {Colors} from '../theme/colors';

interface Props extends DrawerScreenProps<any, any> {}
export const SuccessScreen = ({navigation}: Props) => {
  useEffect(() => {
    navigation.setOptions({
      headerLeft: () => (
        <TouchableOpacity
          style={styles.btnMenu}
          onPress={() => navigation.toggleDrawer()}>
          <Text>Menu</Text>
        </TouchableOpacity>
      ),
    });
  }, [navigation]);

  const {authState} = useContext(AuthContext);

  useEffect(() => {
    if (authState.isLoggedIn) {
      navigation.navigate('StackNavigator');
    }
  }, [authState, navigation]);

  return (
    <SafeAreaView style={[globalStyles.globalMargin, styles.container]}>
      <Image style={styles.logo} source={require('../images/success.gif')} />
      <Text style={[globalStyles.title]}>Â¡You made it!</Text>
      <View style={styles.btnContainer}>
        <TouchableOpacity
          style={[globalStyles.bigButton, styles.btnLogin]}
          onPress={() => navigation.navigate('LoginScreen')}>
          <Text style={globalStyles.bigButtonText}>Go to login</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {flex: 1, justifyContent: 'center', alignItems: 'center'},
  logo: {
    width: 200,
    height: 200,
    resizeMode: 'contain',
  },
  btnContainer: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  inputContainer: {
    width: '100%',
    alignItems: 'center',
    flexDirection: 'column',
  },
  btnLogin: {backgroundColor: Colors.primary},
  btnMenu: {
    marginLeft: 20,
  },
});
