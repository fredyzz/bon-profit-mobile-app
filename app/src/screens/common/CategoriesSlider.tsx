import * as React from 'react';
import {
  Text,
  View,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
} from 'react-native';
import {Colors} from '../../theme/colors';

interface SliderProps {
  categories: Array<string>;
  callBack: (category: string) => void;
  selectedCategory: string | undefined;
}

interface CardProps {
  category: string;
  callBack: (category: string) => void;
  isSelected: boolean | undefined;
}

const CategoryCard = ({category, callBack, isSelected}: CardProps) => (
  <View style={[styles.categoryCard, isSelected && styles.selected]}>
    <TouchableOpacity onPress={() => callBack(category)}>
      <Text style={isSelected && styles.selected}>{category}</Text>
    </TouchableOpacity>
  </View>
);

export const CategoriesSlider = ({
  categories,
  callBack,
  selectedCategory,
}: SliderProps) => {
  return (
    <View style={styles.container}>
      <ScrollView style={styles.scrollView} horizontal={true}>
        {categories.map((category, index) => (
          <CategoryCard
            key={index}
            category={category}
            callBack={callBack}
            isSelected={category === selectedCategory ? true : false}
          />
        ))}
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 140,
    marginTop: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  scrollView: {},
  categoryCard: {
    width: 120,
    height: 120,
    backgroundColor: Colors.white,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.2,
    shadowRadius: 3.84,

    elevation: 4,
    margin: 10,
    borderRadius: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  selected: {
    // shadowColor: Colors.primary,
    // shadowOffset: {
    //   width: 0,
    //   height: 0,
    // },
    // shadowOpacity: 0.8,
    // shadowRadius: 5.84,

    // elevation: 4,
    backgroundColor: Colors.primary,
    color: Colors.white,
  },
});
