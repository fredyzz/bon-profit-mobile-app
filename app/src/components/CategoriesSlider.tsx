import * as React from 'react';
import {View, StyleSheet, ScrollView} from 'react-native';
import {CategoryCard} from './CategoryCard';

interface SliderProps {
  categories: Array<string>;
  callBack: (category: string) => void;
  selectedCategory: string | undefined;
}

export const CategoriesSlider = ({
  categories,
  callBack,
  selectedCategory,
}: SliderProps) => {
  return (
    <View style={styles.container}>
      <ScrollView horizontal={true}>
        {categories.map((category, index) => (
          <CategoryCard
            key={index}
            category={category}
            callBack={callBack}
            isSelected={category === selectedCategory ? true : false}
          />
        ))}
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 140,
    marginTop: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
