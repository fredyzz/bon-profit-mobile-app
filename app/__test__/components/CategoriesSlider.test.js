import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {CategoriesSlider} from '../../src/components/CategoriesSlider';

const mockedCategoriesProp = ['Italian', 'Argentinian'];
const mockedCallBack = jest.fn();
const mockedSelectedCategory = 'Italian';

describe('Given a <CategoriesSlider/> component', () => {
  describe('receiving an array with two categories', () => {
    test('should return two CategotyCard components', () => {
      const tree = renderer
        .create(
          <CategoriesSlider
            categories={mockedCategoriesProp}
            callBack={mockedCallBack}
            selectedCategory={mockedSelectedCategory}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
