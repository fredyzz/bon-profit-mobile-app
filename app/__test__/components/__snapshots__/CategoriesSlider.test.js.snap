// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a <CategoriesSlider/> component receiving an array with two categories should return two CategotyCard components 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 140,
      "justifyContent": "center",
      "marginTop": 20,
    }
  }
>
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 20,
              "elevation": 4,
              "height": 120,
              "justifyContent": "center",
              "margin": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3.84,
              "width": 120,
            },
            Object {
              "backgroundColor": "#95d5b2",
              "color": "white",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="categoryCard-Italian"
        >
          <Text
            style={
              Object {
                "backgroundColor": "#95d5b2",
                "color": "white",
              }
            }
          >
            Italian
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 20,
              "elevation": 4,
              "height": 120,
              "justifyContent": "center",
              "margin": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3.84,
              "width": 120,
            },
            false,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="categoryCard-Argentinian"
        >
          <Text
            style={false}
          >
            Argentinian
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
